packages/typespec-ts/src/modular/helpers/clientHelpers.ts

```diff
@@ -14,7 +14,11 @@ export function getClientParameters(
 
   const params: OptionalKind<ParameterDeclarationStructure>[] = [
     ...parameters
-      .filter((p) => p.type.type !== "constant")
+      .filter(
+        (p) =>
+          p.type.type !== "constant" &&
+          (p.clientDefaultValue === null || p.clientDefaultValue === undefined)
```

Comment:
Really don't remember what is this for, just remember it's a bugfix which has been fixed in TCGC.